graphs/adapter-jgrapht/src/main/kotlin/com/anaplan/engineering/azuki/graphs/adapter/jgrapht/check/HasShortestPathCheck.kt

```diff
@@ -9,20 +9,28 @@ import org.jgrapht.graph.DefaultEdge
 
 class HasShortestPathCheck<V>(
     private val graphName: String,
-    private val from: V,
-    private val to: V,
-    private val shortestPath: List<V>
+    private val shortestPath: List<V>,
 ) : JGraphTCheck, GetShortestPathBehaviour() {
 
-    private val fullShortestPath by lazy {
-        listOf(from) + shortestPath + to
+    private val from = try {
+        shortestPath.first()
+    } catch (e: NoSuchElementException) {
+        error("hasShortestPath must has at least three arguments: graphName, startVertex, endVertex")
     }
+    private val to = shortestPath.last()
```

Comment:
🚫 You'll want to have an assertion earlier on about the length of the list. Also, the error message makes no sense at this level, because the class has only two parameters... what you're referring to is the arguments of a DSL function.

There might be a better place for it, but the easiest way to address this is by using an initializer block, so something like:
```
class HasShortestPathCheck<V>(
    private val graphName: String,
    private val shortestPath: List<V>,
) : JGraphTCheck, GetShortestPathBehaviour() {

    init {
        if (shortestPath.length < 2) { throw LateDetectUnsupportedCheckException("Paths must have at least two endpoints")
    }

    ...
}
```